spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -1,77 +1,127 @@
       end
 
-      it "does not send a notification to a user who has commented in a welcome thread", :aggregate_failures do
-        welcome_thread_article = create(:article, title: "Welcome Thread - v0", published: true, tags: "welcome", user: mascot_account)
-        create(:comment, commentable: welcome_thread_article, commentable_type: "Article", user: user)
+      expect(user.notifications.count).to eq(1)
+    end
+  end
+
+  describe "#send_authentication_notification" do
+    before do
+      allow(Notification).to receive(:send_welcome_notification)
+    end
 
-        expect do
-          sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
-        end.to not_change(user.notifications, :count)
-      end
+    it "does not send notification if user is created less than a day ago" do
+      user = create(:user, :with_identity, identities: ["github"])
+      sidekiq_perform_enqueued_jobs { described_class.new(user.id).send(:send_authentication_notification) }
+      expect(Notification).not_to have_received(:send_welcome_notification)
+    end
 
-      it "does not send a duplicate notification" do
-        2.times { sidekiq_perform_enqueued_jobs { described_class.call(user.id) } }
+    it "generates and sends the appropriate broadcast (twitter)" do
+      user = create(:user, :with_identity, identities: ["github"], created_at: 1.day.ago)
+      sidekiq_perform_enqueued_jobs { described_class.new(user.id).send(:send_authentication_notification) }
+      expect(Notification).to have_received(:send_welcome_notification).with(user.id, twitter_connect_broadcast.id)
+    end
 
-        expect(user.notifications.count).to eq(1)
-      end
+    it "generates and sends the appropriate broadcast (github)" do
+      user = create(:user, :with_identity, identities: ["twitter"], created_at: 1.day.ago)
+      sidekiq_perform_enqueued_jobs { described_class.new(user.id).send(:send_authentication_notification) }
+      expect(Notification).to have_received(:send_welcome_notification).with(user.id, github_connect_broadcast.id)
+    end
 
-      it_behaves_like "unsubscribed from welcome notifications"
+    it "does not send notification if user is authenticated with both services" do
+      user = create(:user, :with_identity, identities: %w[twitter github], created_at: 1.day.ago)
+      sidekiq_perform_enqueued_jobs { described_class.new(user.id).send(:send_authentication_notification) }
+      expect(Notification).not_to have_received(:send_welcome_notification).with(user.id, github_connect_broadcast.id)
     end
 
-    context "when sending a twitter_connect notification" do
-      xit "generates the appropriate broadcast to be sent to a user"
-      xit "it sends a welcome notification for that broadcast"
-      xit "it does not send duplicate welcome notification for that broadcast"
-      xit "does not send a notification to a user who is connected via twitter"
+    # it_behaves_like "unsubscribed from welcome notifications"
```

Comment:
✂️ since the shared example has been removed

Reply:
oops. Removed